compiler/rustc_middle/src/mir/interpret/allocation.rs

```diff
@@ -362,8 +362,9 @@ impl<Prov: Provenance, Extra> Allocation<Prov, Extra> {
     pub fn get_bytes_strip_provenance(
         &self,
         cx: &impl HasDataLayout,
-        range: AllocRange,
+        range: impl Into<AllocRange>,
```

Comment:
Is it worth doing that consistently? Seems a bit ad-hoc right now.
(But also is this even still needed?)

Reply:
yea, this is used in `expect_slice`. I could manually do it there, but it seems convenient?